### ISSUE TITLE
TypeScript symbols is empty inside closure (like jQuery plugins)

### ISSUE THREAD

**Issue by jrieken** [Microsoft/TypeScript#8218]
_From @Ciantic on January 2, 2016 8:38_

I tried developing basic jQuery plugin with TS, looks like the symbols list is totally empty if I use following code:

```
/// <reference path="./typings/tsd.d.ts" />
(function ($: JQueryStatic) {

    interface Anything {
        something : string
    }

    function func() {
    }

})(jQuery);
```

I think it should recgonize the whole file spanning closure and not let it ruin the symbols listing.

_Copied from original issue: Microsoft/vscode#1741_

**Comment by jrieken** [comment]
_From @felixfbecker on January 3, 2016 12:5_

I also think this is annoying. Symbols view only shows symbols in the global scope, also for Javasript. If you use class expressions for Angular, like `.controller('MyController', class MyController { ... })` you will never get any methods of that class show up in the symbol view.

**Comment by jrieken** [comment]
_From @paramquery on April 20, 2016 14:5_

+1

much needed feature.

**Comment by jrieken** [comment]
This is an issue with TypeScript. For JavaScript this actually works a lot better then for TypeScript despite using the same language service...

**Comment by mhegazy** [comment]
@billti had a fix for the JS case. we need to unify the two implementations. https://github.com/Microsoft/TypeScript/issues/7523 tracks some of this work.

**Comment by mhegazy** [comment]
@andy-ms would this be handled by your refactor?

**Comment by andy-ms** [comment]
In the #8958 branch, `Anything` shows up, but `func` doesn't. We had decided that inner functions wouldn't show up if they didn't have any children. I don't know how exactly we should handle this, but maybe the entire module going in a single closure is a pattern we could recognize — although trying to recognize particular patterns like this could pile up to a list of messy special cases.

**Comment by mhegazy** [comment]
i think that is a better place to be in. if that turned out to be another source of complains, may be we should allow all function declarations regardless if they are containers or not.

**Comment by andy-ms** [comment]
Closed by #9220
